packages/synapse-react-client/src/components/EntityViewScopeEditor/EntityViewScopeEditorModal.test.tsx

```diff
@@ -0,0 +1,235 @@
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
+
+    const filesButton = within(dialog).getByRole('checkbox', { name: 'Files' })
+    const foldersButton = within(dialog).getByRole('checkbox', {
+      name: 'Folders',
+    })
+    const tablesButton = within(dialog).getByRole('checkbox', {
+      name: 'Tables',
+    })
+    const datasetsButton = within(dialog).getByRole('checkbox', {
+      name: 'Datasets',
+    })
+
+    const checkboxes = screen.getAllByRole('checkbox')
+    expect(checkboxes).toHaveLength(4)
+
+    expect(filesButton).toBeVisible()
+    expect(foldersButton).toBeVisible()
+    expect(tablesButton).toBeVisible()
+    expect(datasetsButton).toBeVisible()
+
+    await waitFor(() => {
+      expect(filesButton).toBeChecked()
+      expect(foldersButton).toBeChecked()
+      expect(tablesButton).not.toBeChecked()
+      expect(datasetsButton).not.toBeChecked()
+    })
+  })
+
+  it('display error for invalid entity', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    mockUpdateEntity.mockRejectedValue(
+      new SynapseClientError(
+        404,
+        'Error with scope',
+        expect.getState().currentTestName!,
+      ),
+    )
+
+    const saveButton = within(dialog).getByRole('button', { name: 'Save' })
+    await waitFor(() => {
+      expect(
+        within(dialog).getByRole('button', { name: 'Save' }),
+      ).not.toBeDisabled()
+    })
+    await userEvent.click(saveButton)
+    expect(mockOnUpdate).not.toHaveBeenCalled()
+  })
+
+  it('verify no mask options are shown when projectView is passed', async () => {
+    jest.clearAllMocks()
+
+    jest.spyOn(SynapseClient, 'getEntity').mockResolvedValue({
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_PROJECT,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+
+    renderComponent({
+      entityId: mockProjectViewEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
```

Comment:
You never change the visibility of these buttons, so I don't think it's necessary to check `toBeVisible()`

Also, to reduce the amount of duplicate code/simplify reading these tests, you can return the buttons in `renderComponent` so you have them available right away. For example see `setUp` in TableColumnSchemaEditor.integration.test.tsx

Reply:
You can do the same thing with the checkboxes if you use `queryByRole` in the `renderComponent` function. You can then do `expect(filesCheckbox).toBeInTheDocument()` or `expect(filesCheckbox).not.toBeInTheDocument()` depending on the test